### ISSUE TITLE
refactor storytext to use ICU BreakIterator

### ISSUE THREAD

**Issue by Fahad-Alsaidi** [HOST-Oman/scribus#166]
Although sotrytext works find with Latin & Arabic, it fails to deal with east Asian languages ( at least). For example, nextWord() will not work with Thai or CJK, also selectWord() and many other. 

The idea is to use ICU BreakIterator which include:
1.     Character Boundary
2.     Word Boundary
3.     Line-break Boundary
4.     Sentence Boundary

To get a good performance as well as result we need:
1. know what is current Locale.
2. create BreakIterator types somewhere general to use it in different places.
3. delete unused functions ( surprisingly they are a lot)

**Comment by Fahad-Alsaidi** [comment]
testing example
[langs-text.sla.gz](https://github.com/HOST-Oman/scribus/files/237940/langs-text.sla.gz)
.

**Comment by Fahad-Alsaidi** [comment]
I welcome any feedback before I start refactoring.

**Comment by khaledhosny** [comment]
I’m not sure what you mean by point 3, can you elaborate on what code you want to remove? I don’t want to do big refactoring at this stage since we are trying to stabilize the code to merge it with trunk, so I’d try to limit the changes to what fixes a real issue.

ICU break iterators are quite expensive so we need to cache them. In my experience what locale you pick does not seem to matter, so try to pass an empty locale and see what happens. For caching the iterator, may be you can declare it static so it will be created only once (this might be thread unsafe, though. Needs checking).

**Comment by MrB74** [comment]
khaledhosny: You are correct, we/you need to stabilise CTL so we can bring it into SVN after 1.5.2 is released. Any major enhancements or changes should be kept until afterwards for merging in 1.5.3 or later.

**Comment by Fahad-Alsaidi** [comment]
@khaledhosny @MrB74  I mean by "delete unused functions " just deleting functions that never called. There is no harm to delete dead code.

**Comment by khaledhosny** [comment]
Even if it is unused code, we can postpone such (mostly cosmetic) cleanup after the merge.

**Comment by Fahad-Alsaidi** [comment]
@khaledhosny ok I will not touch them.
